### DISCUSSION TITLE
doris查询paimon表报错：Unknown type value: UNSUPPORTED

### DISCUSSION THREAD

**Discussion by rulerhuang** [apache/doris#25174] [A - General / Q&A] [answered]
【环境信息】
- doris-2.0.2
- paimon 0.5
- flink 1.16
- hadoop 3.3.6

【复现步骤】
1、创建paimon catalog：
CREATE CATALOG `paimon_hdfs` PROPERTIES (
    "type" = "paimon",
    "warehouse" = "hdfs://x.x.x.x:9000/root/data/flink-1.16/paimon/warehouse",
    "metadata_refresh_interval_sec" = "20"
);

2、可以使用该catalog查看表结构
mysql> show tables;
+--------------------+
| Tables_in_dev_lake |
+--------------------+
| dim_test_cases     |
+--------------------+
1 row in set (0.82 sec)

mysql> desc dim_test_cases;
+----------------+--------+------+------+---------+-------+
| Field          | Type   | Null | Key  | Default | Extra |
+----------------+--------+------+------+---------+-------+
| issue_id       | BIGINT | Yes  | true | NULL    |       |
| project_id     | BIGINT | Yes  | true | NULL    |       |
| jira_key       | TEXT   | Yes  | true | NULL    |       |
| summary        | TEXT   | Yes  | true | NULL    |       |
| status         | TEXT   | Yes  | true | NULL    |       |
| execution_mode | TEXT   | Yes  | true | NULL    |       |
| priority       | TEXT   | Yes  | true | NULL    |       |
+----------------+--------+------+------+---------+-------+
7 rows in set (0.00 sec)

3、但是select表时直接报错
mysql> select * from dim_test_cases limit 3;
ERROR 1105 (HY000): errCode = 2, detailMessage = (10.120.18.252)[CANCELLED][INTERNAL_ERROR]RuntimeException: Unknown type value: UNSUPPORTED
0. /root/src/doris-2.0/be/src/common/stack_trace.cpp:302: StackTrace::tryCapture() @ 0x000000000ba1f197 in /root/packages/apache-doris-2.0.2-bin-x64/be/lib/doris_be
1. /root/src/doris-2.0/be/src/common/stack_trace.h:0: doris::get_stack_trace[abi:cxx11]() @ 0x000000000ba1d72d in /root/packages/apache-doris-2.0.2-bin-x64/be/lib/doris_be
2. /var/local/ldb-toolchain/bin/../lib/gcc/x86_64-lin

**Comment by rulerhuang** [comment]
怀疑是doris不支持paimon的BIGINT类型

**Comment by zy-kkk** [comment]
Can you provide the table creation statement for this table in Paimon?

**Reply by rulerhuang** [reply]
CREATE TABLE if NOT EXISTS `dim_test_cases` (
            `issue_id`              BIGINT NOT NULL COMMENT 'issue id',
            `project_id`            BIGINT NOT NULL COMMENT '项目ID',
            `jira_key`              VARCHAR(128) NOT NULL COMMENT '测试计划 jira key',
            `summary`               VARCHAR(256) NOT NULL COMMENT '测试计划名称',
            `status`                VARCHAR(128) NOT NULL COMMENT '状态',
            `execution_mode`        VARCHAR(32) NOT NULL COMMENT '测试用例执行方式',
            `priority`              VARCHAR(32) NOT NULL COMMENT '测试用例优先级',
          PRIMARY KEY (`issue_id`) NOT ENFORCED
        );

**Reply by rulerhuang** [reply]
直接select VARCHAR类型的字段是OK的，但是select issue_id 或 select project_id 就报错了

**Comment by liugddx** [comment]
use paimon 0.4 and try it again.

**Reply by rulerhuang** [reply]
换成 paimon 0.4 也是一样的报错

**Comment by change1224** [comment]
同样的问题，解决了吗？

**Reply by rulerhuang** [reply]
看样子是已知bug，paimon版本换了也没用，就是类型映射有bug

**Comment by xiaolin84250** [comment]
![image](https://github.com/apache/doris/assets/38107489/b0ec66dc-7712-445f-b521-e44114dde3f7)
![image](https://github.com/apache/doris/assets/38107489/871aeff7-f03d-4523-9f62-7c5db34dddc6)
![image](https://github.com/apache/doris/assets/38107489/8e9d3b6f-4481-4c9e-a95d-d337964a09a6)

我也遇到了这个问题，目前还没有解决方案吧？

**Reply by xiaolin84250** [reply]
查询string类型的单字段我查了能查出来

**Reply by DongLiang-0** [reply, marked as answer]
This PR has been modified and you need to wait for the upgrade to doris 2.0.3.https://github.com/apache/doris/pull/24985

**Reply by xiaolin84250** [reply]
thank!!!

**Reply by morningman** [reply]
you can also download this binary:https://apache-doris-releases.oss-accelerate.aliyuncs.com/apache-doris-2.0.3-rc01-bin-x64.tar.gz

**Reply by rulerhuang** [reply]
excellent !!!
